### PR TITLE
Pain

### PR DESCRIPTION
**Pull request by velimd**
### JIRA link ###
https://tools.hmcts.net/jira/browse/NFDIV-514

### Change description ###
pain,
first commit is clean and is basically moving the get partner to common page.
second commit is basically making all content stuff a function and loading it in common content so all the content is done in one go.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/app/case/answers/getAnswerRows.ts [hunk 1]
```diff
@@ -24,7 +24,8 @@ export const getAnswerRows = function (section: Sections): GovUkNunjucksSummary[
     .filter(step => step.showInSection === section)
     .flatMap(step => {
       const fieldKeys = Object.keys(step.form.fields);
-      const stepContent = { ...this.ctx, ...step.generateContent({ isDivorce, partner, formState })[language] };
+      console.log(language);
+      const stepContent = { ...this.ctx, ...step.generateContent({ isDivorce, partner, formState }) };
```
Comment (velimd) [inline src/main/app/case/answers/getAnswerRows.ts:28]: this part was just to push and show what the content files now look like. ill fix this after if the way is approved or requires changes.

File: src/main/steps/common/common.content.ts [hunk 2]
```diff
@@ -147,4 +152,41 @@ const cy: typeof en = {
...
+  let partner;
+  if (!isDivorce) {
+    partner = commonTranslations['civilPartner'];
+  } else if (selectedGender === Gender.MALE) {
+    partner = commonTranslations['husband'];
+  } else if (selectedGender === Gender.FEMALE) {
+    partner = commonTranslations['wife'];
+  } else {
+    partner = commonTranslations['partner'];
+  }
```
Comment (linusnorton) [inline src/main/steps/common/common.content.ts:172]: make that a function so you don't need to use a let

**Review by linusnorton — APPROVED** [review]

File: src/main/steps/common/common.content.ts [hunk 3]
```diff
@@ -147,4 +152,52 @@ const cy: typeof en = {
...
+    Object.assign(content, pageContent({ commonTranslations, ...content } as CommonContent));
+  }
+
+  return {
+    ...commonTranslations,
+    ...content,
+  };
+};
+
+const getPartnerContent = (translations, selectedGender: Gender, isDivorce = true): string => {
```
Comment (linusnorton) [inline src/main/steps/common/common.content.ts:182]:
```suggestion
const getPartnerContent = (translations, selectedGender: Gender, isDivorce): string => {
```

File: src/main/steps/common/common.content.ts [hunk 4]
```diff
@@ -147,4 +152,52 @@ const cy: typeof en = {
...
+  const commonTranslations: typeof en = language === 'en' ? en : cy;
+  const selectedGender = formState?.gender as Gender;
+  const partner = getPartnerContent(commonTranslations, selectedGender, isDivorce);
+
+  const content = {
+    partner,
+    language,
+    isDivorce,
+    formState,
+  };
```
Comment (linusnorton) [inline src/main/steps/common/common.content.ts:170]:
```suggestion
    ...commonTranslations
  };
```

File: src/main/steps/common/common.content.ts [hunk 5]
```diff
@@ -147,4 +152,52 @@ const cy: typeof en = {
...
+
+  if (pageContent) {
+    Object.assign(content, pageContent({ commonTranslations, ...content } as CommonContent));
```
Comment (linusnorton) [inline src/main/steps/common/common.content.ts:173]:
```suggestion
    Object.assign(content, pageContent(content));
```

File: src/main/steps/common/common.content.ts [hunk 6]
```diff
@@ -147,4 +152,52 @@ const cy: typeof en = {
...
+  }
+
+  return {
```
Comment (linusnorton) [inline src/main/steps/common/common.content.ts:176]:
```suggestion
  return content;
```

**Review by SebFlippence — APPROVED** [review]

**Pull request merged**